ARUHA-838: Added validation of empty consumer group

v-stepanov opened the pull request:
https://techjira.zalando.net/browse/ARUHA-838

adyach commented on src/main/java/org/zalando/nakadi/domain/SubscriptionBase.java:
```diff
@@ -29,6 +29,7 @@
     private Set<String> eventTypes;
 
     @NotNull
+    @Size(min = 1, message = "must contain at least one character")
```
I thought it should be something that make sense, like 3 chars, because you can form the work from them, like `cat` :)

rcillo replied:
What if users want to have consumer groups with a single char, like "1", "2", etc. They would be forced to do something like "001" which could be frustranting. I'm fine with single char.

adyach replied:
Such consumer group is meaningless, that's why I want to forbid creating them.

v-stepanov replied:
For me 1 and 2 chars can also make sense. And 3 chars is not a guarantee that it will make sense :)
So I would prefer to keep it as it is.

adyach replied:
Maybe then adjusting max value? http://docs.oracle.com/javaee/7/api/javax/validation/constraints/Size.html#max--

v-stepanov replied:
@adyach why do you want to adjust max value?

adyach replied:
@v-stepanov because the default max value is 2147483647, in my opinion it is too much for consumer group. And it is good practice to validate borders of the input.

v-stepanov replied:
Well, in general yes, but we don't really validate the max size of properties anywhere except the event-type name. So if we decide to do that we should add that for all fields and as a separate task. Doing it right now for this only field will not bring much value as for me.

rcillo commented:
@v-stepanov do you think it's worth it to update the API by adding something like `minLength` to the definition of `consumer_group` string?

v-stepanov commented:
@rcillo I added minLength to API now https://github.com/zalando/nakadi/pull/680/commits/6941085f9bbc24d5261a6ee55e52b095a3eed2b5
also fixed some errors in swagger file.

rcillo commented:
👍

v-stepanov commented:
👍

rcillo commented:
👍
